### PR TITLE
Implementing CLDR testing using `unitsTest.txt`

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: experimental/unitsconversion/tests/unitsTest.txt [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# Test data for unit conversions
```
Comment (sffc) [inline experimental/unitsconversion/tests/unitsTest.txt:1]: By convention, this should be located at `experimental/unitsconversion/tests/data/unitsTest.txt`

File: experimental/unitsconversion/Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,7 @@ databake = { workspace = true, optional = true, features = ["derive"]}
 displaydoc = { version = "0.2.3", default-features = false }
 icu_locid = { workspace = true }
 icu_provider = { workspace = true, features = ["macros"] }
+num = "0.4.1"
```
Comment (sffc) [inline experimental/unitsconversion/Cargo.toml:28]: Issue: dev dependency

File: experimental/unitsconversion/tests/units_test.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        let ten = BigRational::from_integer(10.into());
+        let exponent_part = ten.pow(exponent_part);
+        rational_part *= exponent_part;
+        return Some(rational_part);
+    }
+
+    #[test]
+    fn test_conversion() {
+        // let provider = icu_unitsconversion::provider::UnitsInfoV1;
+
```
[the remark this answers is not in the document]
Comment (sffc) [reply experimental/unitsconversion/tests/units_test.rs:54]: What are you testing?

If you are testing data generation, the test should go in datagen

If you are testing functionality, you should implement the functionality and hook up compiled data.

**Review by sffc — CHANGES_REQUESTED** [review]

File: experimental/unitsconversion/tests/units_test.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs:54]: For now, I testing that all the input and output units are parsable. in the next PR, I will use the same test to test the convertibility and conversion.

I need to access the  `UnitsInfoV1` in order to create a parser right now.

[the remark this answers is not in the document]
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs:54]: from @robertbastian , look at the translitrator, use BakedData

File: experimental/unitsconversion/tests/unitsTest.txt (hunk 1 above)
Comment (younies) [reply experimental/unitsconversion/tests/unitsTest.txt]: done

File: experimental/unitsconversion/Cargo.toml (hunk 2 above)
Comment (younies) [reply experimental/unitsconversion/Cargo.toml:29]: done

File: experimental/unitsconversion/tests/data/baked/macros/units_info_v1.rs.data [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
...\0\x13\0\0\0\x14\0\0\0\x15\0\0\0\x01\x03\0:\0\x02\0\0S\0\xFE\0\0|\0\xD0\x80I\x06\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x07\0\0\0\n\0\0\0\x0B\0\0\0\x03\0\0S\0a\t\xEC\x07\x02\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\t\0\0\0\x0B\0\0\0\x0C\0\0\0\x01\x03\0:\0\x85 \xB4\x02P\xC3\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x07\0\0\0\x08\0\0\0\t\0\0\0\x01\x03\0:\0\xE8\x03\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x14\0\0\0\x15\0\0\0\x16\0\0\0\x17\0\0\0\x01\x03\0:\0\x02\0\0S\0\xFD\0\0|\0\xFF\0\0\x02\0\x01\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x0F\0\0\0\x10\0\0\0\x11\0\0\0\x12\0\0\0\x01\x03\0:\0\x02\0\0S\0\xFD\0\0|\0\x01\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x14\0\0\0\x15\0\0\0\x16\0\0\0\x17\0\0\0\x01\x03\0:\0\x02\0\0S\0\xFE\0\0|\0\xFF\0\0\x02\0\x01\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x08\0\0\0\t\0\0\0\n\0\0\0\x01\0\0|\0\x80:\t\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x08\0\0\0\t\0\0\0\n\0\0\0\x01\0\0|\0\x80:\t\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x07\0\0\0\t\0\0\0\n\0\0\0\x01\0\0S\0w\x04\xE2\x04\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x06\0\0\0\x07\0\0\0\x08\0\0\0\x01\0\0\x96\0\x01\x01\0\x01\0\0\0\x05\0\0\0\0\0\0\0\x05\0\0\0\x06\0\0\0\x07\0\0\0\x08\0\0\0\x01\0\0\x96\0\x01\x01\0\x01") } };
```
Comment (younies) [inline experimental/unitsconversion/tests/data/baked/macros/units_info_v1.rs.data:14]: @robertbastian : how to read this in `experimental/unitsconversion/tests/units_test.rs`

Comment (robertbastian) [reply experimental/unitsconversion/tests/data/baked/macros/units_info_v1.rs.data:14]: `icu_unitsconversion::provider::Baked::SINGLETON_UNITS_INFO_V1`

Comment (younies) [reply experimental/unitsconversion/tests/data/baked/macros/units_info_v1.rs.data:14]: Thx :)

File: experimental/unitsconversion/tests/units_test.rs [hunk 5]
```diff
@@ -0,0 +1,155 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(test)]
+mod tests {
+    use core::str::FromStr;
+    use icu_unitsconversion::measureunit::MeasureUnitParser;
+    use num::BigRational;
+    use std::collections::HashSet;
```
Comment (younies) [inline experimental/unitsconversion/tests/units_test.rs:10]: Is it okay to use `std` in testing ?

Comment (robertbastian) [reply experimental/unitsconversion/tests/units_test.rs:10]: yes

File: experimental/unitsconversion/tests/units_test.rs [hunk 6]
```diff
@@ -0,0 +1,155 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(test)]
+mod tests {
```
Comment (robertbastian) [inline experimental/unitsconversion/tests/units_test.rs:6]: not requires in a `tests/` test

File: experimental/unitsconversion/tests/units_test.rs [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            "to-jp",
+            "month-person",
+            "year-person",
+            "decade",
+        ]
+        .iter()
+        .cloned()
+        .collect();
+        for test in tests {
+            // TODO: remove this line after fixing the parser.
```
Comment (robertbastian) [inline experimental/unitsconversion/tests/units_test.rs:140]: do you have an issue?

File: experimental/unitsconversion/tests/units_test.rs [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+                    _category: parts[0].to_string(),
+                    input_unit: parts[1].to_string(),
+                    output_unit: parts[2].to_string(),
+                    _result: get_rational(parts[4]).unwrap(),
+                }
+            })
+            .collect();
+
+        let parser = MeasureUnitParser::from_payload(
+            &icu_unitsconversion::provider::Baked::SINGLETON_UNITS_INFO_V1.units_conversion_trie,
```
Comment (robertbastian) [inline experimental/unitsconversion/tests/units_test.rs:82]: are you following the transliteration model with baked data in the `tests/` folder, or are you properly hooking up baked date with a `icu_units_conversion_data` crate? In the former case, this is not correct, in the latter you don't need the files in `tests/data`

File: experimental/unitsconversion/tests/units_test.rs (hunk 6 above)
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs:6]: done

File: experimental/unitsconversion/tests/units_test.rs (hunk 7 above)
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs]: done: https://github.com/unicode-org/icu4x/issues/4461

File: experimental/unitsconversion/tests/units_test.rs (hunk 8 above)
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs]: As I saw at the transliteration, they have copied the baked data into tests.

Can I read the data directly from the Datagen place?

Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs]: Actually, we can just access the data directly.

File: experimental/unitsconversion/tests/units_test.rs [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+            continue;
+        }
+        let input_unit = parser.try_from_identifier(test.input_unit.as_str());
+        let output_unit = parser.try_from_identifier(test.output_unit.as_str());
+
+        assert!(input_unit.is_ok());
+        assert!(output_unit.is_ok());
+    }
+
+    // let parser = MeasureUnitParser::from_payload(&BakedDataProvider);
```
Comment (robertbastian) [inline experimental/unitsconversion/tests/units_test.rs:151]: ?

**Review by robertbastian — APPROVED** [review]

File: experimental/unitsconversion/tests/units_test.rs (hunk 9 above)
Comment (younies) [reply experimental/unitsconversion/tests/units_test.rs:151]: I forgot to delete, done

**Review by sffc — APPROVED** [review]

**Pull request merged**